### PR TITLE
Match case/Docker

### PR DESCRIPTION
**Pull request by traikodinev**
This PR:

- Adds the exercises for `match`-`case`.
- Adds a docker image for exercises, which has `python 3.11` so that learners don't need to update their python version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.11
+
+ADD requirements.txt requirements.txt
+RUN pip install -r requirements.txt
+
+RUN echo > /root/.bashrc
+RUN echo "PS1='🐳\[\033[01;32m\]\u@:\[\033[01;34m\]\W\[\033[00m\] \$ '" >> /root/.bashrc
+RUN echo "alias ls='ls --color=auto'" >> /root/.bashrc
```
Comment (davidmillican) [inline Dockerfile:8]: What's this doing? Can we add a comment to explain these lines?

**Review by davidmillican — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (traikodinev) [reply Dockerfile:8]: The first line modifies your prompt, the second makes `ls` by default have colors for folders and files. I will add a comment.

Screenshot 2023-08-01 at 09 10 23

Comment (davidmillican) [reply Dockerfile:8]: Is this standardised across our images? Or just for this Python one?

Comment (traikodinev) [reply Dockerfile:8]: That's a good point. There's actually nothing in the other images.

I suppose I can remove that from here to make it more consistent.

Screenshot 2023-08-01 at 09 33 06

**Comment by davidmillican** [comment]
No, I'm happy for you to include it! Just add a comment to say what it
does, and then perhaps we should start to standardise on things that seem
good. Your version is much easier to read.

**Pull request merged**